lib/src/_helpers/utils.ts

```diff
@@ -6,3 +8,13 @@ export function arrayIncludes<T>(array: T[] | readonly T[], itemOrItems: T | T[]
 
   return array.indexOf(itemOrItems) !== -1;
 }
+
+export const onSpaceOrEnter = (innerFn: () => void) => (e: React.KeyboardEvent) => {
+  if (e.key === 'Enter' || e.key === ' ') {
+    innerFn();
+
+    // prevent any side effects
+    e.preventDefault();
+    e.stopPropagation();
```

Comment:
I think that we should be cautious about the usage of stopPropagation. I think that it would be great to add a comment on the "why?" we use it here for our future self. https://css-tricks.com/dangers-stopping-event-propagation/